gunrock/app/sm/sm_app.cu

```diff
@@ -48,23 +48,27 @@ cudaError_t UseParameters(util::Parameters &parameters) {
  * @param[in]  query_col_indices CSR-formatted graph input query column indices
  * @param[in]  num_runs          Number of runs to perform SM
  * @param[out] subgraphs         Return number of subgraphs
+ * @param[out] list_subgraphs    Return list of subgraphs
+ * @param[in]  device            input and output target device, by default CPU
  * \return     double            Return accumulated elapsed times for all runs
  */
 double sm(
-    const int            num_nodes,
-    const int            num_edges,
-    const int           *row_offsets,
-    const int           *col_indices,
-    const int            num_query_nodes,
-    const int            num_query_edges,
-    const int           *query_row_offsets,
-    const int           *query_col_indices,
-    const int            num_runs,
-          int           *subgraphs)
+    const int                 num_nodes,
+    const int                 num_edges,
+    const int                *row_offsets,
+    const int                *col_indices,
+    const int                 num_query_nodes,
+    const int                 num_query_edges,
+    const int                *query_row_offsets,
+    const int                *query_col_indices,
+    const int                 num_runs,
+    unsigned long            *subgraphs,
+    unsigned long            *list_subgraphs,
+    gunrock::util::Location   device = gunrock::util::HOST)
```

Comment:
`device = HOST` is confusing. device generally implies GPU.
Can we change this to `target`, or `allocated_on`?

Reply:
Sure, I can change the name to `allocated_on`